RUFAS/current_weather.py

```diff
@@ -0,0 +1,107 @@
+        Maximum air temperature for the day (C).
+    daylength: float, optional, default=None
+        Length of time from sunup to sundown on the day (hours).
+    annual_mean_air_temperature: float, optional, default=None
+        Average annual air temperature for the year (C).
+    snow_fall: float, default=0.0
+        Amount of snow that falls on the day (mm).
+    rainfall: float, default=0.0
+        Amount of rainfall that occurs on the day (mm).
+    irrigation: float, default=0.0
+        Amount of irrigation that is applied to the field on that day (mm).
+    precipitation: float, default=0.0
+        Amount of precipitation that occurs on the day (mm).
+
+    Notes
+    -------
+    _deg_trig and _determine_daylength are more of temporary methods that approximately estimates the day length, this
+    will be revisited for a more accurate implementation post v1
+
+    """
+    incoming_light: Optional[float] = None
+    min_air_temperature: Optional[float] = None
+    mean_air_temperature: Optional[float] = None
+    max_air_temperature: Optional[float] = None
+    daylength: Optional[float] = None
+    annual_mean_air_temperature: Optional[float] = None
+    snow_fall: float = 0.0
+    rainfall: float = 0.0
+    irrigation: float = 0.0
+    precipitation: float = 0.0
+
+    def __post_init__(self):
+        """Sets precipitation as snow_fall or rainfall depending on mean air temperature"""
+        if self.mean_air_temperature >= 0:
+            self.rainfall = self.precipitation
+        else:
+            self.snow_fall = self.precipitation
+
+    @staticmethod
+    def determine_daylength(month: int) -> int:
+        """
+        Approximate day length of the month by using data from Madison, WI.
+
+        Parameters
+        ----------
+        month: int
+            Month of the year.
+
+        Returns
+        -------
+        int
+            Day length of the month (hour).
+
+        References
+        ----------
+        https://sunrise.maplogs.com/dane_county_wi_usa.15449.html
+        """
+        daylength = [9, 10, 11, 13, 14, 15, 15, 15, 13, 12, 10, 9]
+        return daylength[month-1]
+
+    @staticmethod
+    def date_conversion_month(time: Time) -> int:
+        """
+        Converts the day number into the corresponding month of the year.
+
+        Parameters
+        ----------
+        time: Time
+            Time object containing the current time of the simulation.
+
+        Returns
+        -------
+        int
+            The corresponding month of the year.
+
+        """
+        days = [31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365]
+        if is_leap_year(time.calendar_year):
+            days = [31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366]
+        prev_month = 0
+        for day in days:
+            if prev_month < time.day <= day:
+                return days.index(day) + 1
+            else:
+                prev_month = day
```

Comment:
The code can be written more clearly. The name of the function is vague. This function has nothing to do with the weather; it belongs to utils.
```suggestion
  @staticmethod
  def day_to_month_conversion(time: Time) -> int:
      """
      Converts the day number into the corresponding month of the year.
      
      Parameters
      ----------
      time: Time
          Time object containing the current time of the simulation.
      
      Returns
      -------
      int
          The corresponding month of the year (1 for January, 2 for February, etc.).
      """
      # Define the cumulative day counts for non-leap and leap years
      non_leap_cumulative_days_in_months = [31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365]
      leap_cumulative_days_in_months = [31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366]
      
      cumulative_days_in_months = leap_cumulative_days_in_months if is_leap_year(time.calendar_year) else non_leap_cumulative_days_in_months 
      
      for month, day_count in enumerate(cumulative_days_in_months ):
          if time.day <= day_count:
              return month + 1 

```

Reply:
Suggested code changes have been added, and method has been moved to `Utility`.

Important note: I moved the test for this method to a new file `test_util.py`. In order to avoid creating more issues in the same vein as #649, I moved all the other existing tests for `Utility` from `test_misc.py` to `test_util.py`. If this change is undesirable it can be reverted.

Reply:
Sounds good, `test_misc.py` needed to be broken down anyways.